code/game/objects/items/weapons/tools.dm

```diff
@@ -348,22 +348,31 @@
 			return
 		tank.armed = 1
```

Comment:
This is not reset in the "No" branch of the code.

Reply:
Alternatively, only set it in the "Yes" branch of the code. It's cleaner that way.